File: Algorithms/Search Algorithms/linearSearch.java
import java.util.Scanner;

class linearSearch { 
    // This function returns index of element x in arr[] 
    static int search(int arr[], int n, int x) 
    { 
        for (int i = 0; i < n; i++) { 
            // Return the index of the element if the element 
            // is found 
            if (arr[i] == x) 
                return i; 
        } 
  
        // return -1 if the element is not found 
        return -1; 
    } 
  
    public static void main(String[] args) 
    { 
    	Scanner sc= new Scanner(System.in);
    	System.out.print("Enter the number of elements to be inserted in an array:");
    	int num= sc.nextInt();
    	int[] arr = new int[num];
    	System.out.println("Enter the elements to be inserted:");
    	for(int i=0;i<num;i++) {
    		arr[i] = sc.nextInt(); 
    	}
        int len = arr.length; 
        System.out.print("Enter the integer to be searched in an array:"); 
        int x = sc.nextInt(); 
  
        int index = search(arr, len, x); 
        if (index == -1) 
            System.out.println("Element is not present in the array"); 
        else
            System.out.println("Element found at position " + index); 
    } 
}

File: Algorithms/Search Algorithms/linear search.cpp
Linear Search in C++

What is a linear search in C++ ?
Linear search is one of the simplest algorithm of data structure. 
In the process of linear search, the element to be searched is compared with every element of the list one by one until the element that is to be searched is found. 
If the element is not found till the end this means that the element is not present in the list. 
This can be performed on both the sorted list or the unsorted list.


Algorithm to implement linear search in C++
1.Read the element to be search from the user.
2.Compare the search element with the first element in the list.
3.If they both match,terminate the function.
4.Else compare the search element with the next element in the list.
5.Repeat steps 3 and 4 until the element to be search is found.


Program to implement linear search algorithm in C++

#include<bits/stdc++.h>
using namespace std;

int main()
{
    int i,n,key;
    
    cout<<"Enter the size of the array: ";
    cin>>n;
    int *arr = new int(n);  
    
    cout<<"Enter the array elements: ";
    for(i=0;i<n;i++)
    cin>>arr[i];
    
    cout<<"Enter the key to be searched: ";
    cin>>key;
    
    for(i=0;i<n;i++)    //Searching element using linear search.
    {
      if(key==arr[i])
      {
        cout<<"The key element is found at index : "<<i+1;
        break;
      }
    }
    if(i==n)
    cout<<"Element not found";
}


Time Complexity
For Linear Search
Best
O(1)